test/acceptance/keywords/screenshots.robot

```diff
@@ -79,3 +79,72 @@ Capture page screenshot with escaped braces
     Should Be Equal    ${file}    ${OUTPUTDIR}${/}screenshot-and-index${/}brackets-{index}-name.png
     ${file} =    Capture Page Screenshot    ${OUTPUTDIR}${/}screenshot-and-index${/}brackets-{{index-name.png
     File Should Exist    ${OUTPUTDIR}${/}screenshot-and-index${/}brackets-{index-name.png
+
+Capture page screenshot computed name is unique
+    [Documentation]  verify that computed filenames are unique
+    [Tags]     issue-502
+    [Setup]    run keywords
+    ...  Remove files  ${OUTPUTDIR}/unique-screenshot-*.png
+    ...  AND  Touch    ${OUTPUTDIR}/unique-screenshot-1.png
+    ...  AND  Touch    ${OUTPUTDIR}/unique-screenshot-2.png
+    ...  # unique-screenshot-3 is purposely left out
+    ...  AND  Touch    ${OUTPUTDIR}/unique-screenshot-4.png
+
+
+    # number 3.
+    ${expected}=  set variable    ${OUTPUTDIR}/unique-screenshot-3.png
+    ${actual}=    Capture page screenshot  ${OUTPUTDIR}/unique-screenshot-{index}.png
+    Should be equal  ${actual}  ${expected}  values=False
+    ...  msg=Expected screenshot to be named '${expected}' but it was '${actual}'
+    File Should Exist    ${expected}
+
+    # since screenshot 4 exists, the next should be screenshot 5.
+    ${expected}=  set variable    ${OUTPUTDIR}/unique-screenshot-5.png
+    ${actual}=  Capture page screenshot  ${OUTPUTDIR}/unique-screenshot-{index}.png
+    Should be equal  ${actual}  ${expected}  values=False
+    ...  msg=Expected screenshot to be named '${expected}' but it was '${actual}'
+    File Should Exist    ${expected}
+
+Capture page screenshot advanced formatting name is unique
+    [Documentation]  verify that computed filenames are unique
+    [Tags]     issue-502
+    [Setup]    run keywords
+    ...  Remove files  ${OUTPUTDIR}/advanced-screenshot-*.png
+    ...  AND  Touch    ${OUTPUTDIR}/advanced-screenshot-002.png
+    ...  AND  Touch    ${OUTPUTDIR}/advanced-screenshot-003.png
+    ...  # advanced-screenshot-4 is purposely left out
+    ...  AND  Touch    ${OUTPUTDIR}/advanced-screenshot-005.png
+
+    ${expected}=  set variable    ${OUTPUTDIR}/advanced-screenshot-001.png
+    ${actual}=    Capture page screenshot  ${OUTPUTDIR}/advanced-screenshot-{index:03}.png
+    Should be equal  ${actual}  ${expected}  values=False
+    ...  msg=Expected screenshot to be named '${expected}' but it was '${actual}'
+    File Should Exist    ${expected}
+
+    # since screenshot 3 exists, the next should be screenshot 4.
```

Comment:
Should the comment say: `since screenshot 2 and 3 exists, the next should be screenshot 4`